utils_for_test.go

```diff
@@ -84,6 +85,71 @@ func writeFile(t *testing.T, prefix, contents string) string {
 	return f.Name()
 }
 
+type debugThrottler struct {
+	sync.Mutex
+	credits float64
+}
+
+func (d *debugThrottler) IsAllowed(operation string) bool {
+	const (
+		creditsPerDebugTrace = 1
+	)
+	d.Lock()
+	defer d.Unlock()
+	if d.credits >= creditsPerDebugTrace {
+		d.credits -= creditsPerDebugTrace
+		return true
+	}
+	return false
+}
+
+func newDebugThrottler(credits float64) *debugThrottler {
+	return &debugThrottler{credits: credits}
+}
+
+type debugTraceCounter struct {
+	sync.Mutex
+	numDebugSpans int
+	max           int
+	t             *testing.T
+}
+
+func newDebugTraceCounter(numDebugTraces int, t *testing.T) jaeger.Reporter {
```

Comment:
`t *testing.T` should be the first argument

Reply:
👍